core/src/main/java/io/questdb/cairo/CairoEngine.java

```diff
@@ -95,6 +97,20 @@ public int getBusyWriterCount() {
         return writerPool.getBusyCount();
     }
 
+    public void migrateNullFlag(TableWriter writer, CairoSecurityContext cairoSecurityContext, CharSequence tok) {
+        TableReader reader = getReader(cairoSecurityContext, tok);
+        TableReaderMetadata readerMetadata = (TableReaderMetadata) reader.getMetadata();
+        if (readerMetadata.getVersion() < 416){
+            for (int i = 0; i < reader.getColumnCount(); i++) {
```

Comment:
this is minor, but all loops are like this:

```
for (int i =0, n = something.count(); i < n; i++) {
```

this is in case `getColumnCount` is relatively heavy